lib/internal/readline/promises.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ArrayPrototypePush,
+  Promise,
+  PromiseResolve,
+} = primordials;
+
+const { CSI } = require('internal/readline/utils');
+const { validateInteger } = require('internal/validators');
+const { isWritable } = require('internal/streams/utils');
+const { codes: { ERR_INVALID_ARG_TYPE } } = require('internal/errors');
+
+const {
+  kClearToLineBeginning,
+  kClearToLineEnd,
+  kClearLine,
+  kClearScreenDown,
+} = CSI;
+
+class Readline {
+  #stream;
+  #todo = [];
+
+  constructor(stream) {
```

Comment:
Maybe you can add an `autoCommit` flag that automically calls `.commit()` in the next microtick.

Reply:
Done in https://github.com/nodejs/node/pull/37947/commits/a8efd3130cec610bc19c42b72fda0f3c362a1bb7.